tests/prevrandao_test.go

```diff
@@ -0,0 +1,71 @@
+package tests
+
+import (
+	"context"
+	"github.com/Fantom-foundation/go-opera/tests/contracts/prevrandao"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"testing"
+)
+
+func TestPrevRandao(t *testing.T) {
+	net, err := StartIntegrationTestNet(t.TempDir())
+	if err != nil {
+		t.Fatalf("Failed to start the fake network: %v", err)
+	}
+	defer net.Stop()
+
+	// Deploy the contract.
+	contract, _, err := DeployContract(net, prevrandao.DeployPrevrandao)
+	if err != nil {
+		t.Fatalf("failed to deploy contract; %v", err)
+	}
+
+	// Collect the current PrevRandao fee from the head state.
+	receipt, err := net.Apply(contract.LogCurrentPrevRandao)
+	if err != nil {
+		t.Fatalf("failed to log current prevrandao; %v", err)
+	}
+
+	if len(receipt.Logs) != 1 {
+		t.Fatalf("unexpected number of logs; expected 1, got %d", len(receipt.Logs))
+	}
+
+	entry, err := contract.ParseCurrentPrevRandao(*receipt.Logs[0])
+	if err != nil {
+		t.Fatalf("failed to parse log; %v", err)
+	}
+	fromLog := entry.Prevrandao
+
+	// Collect the prevrandao from the block header.
```

Comment:
This comment is wrong. You are actually requesting the prevrandao from the archive. If you would request it from the block header, you would expect it to be different.